src/Components/DrillPage.js

```diff
@@ -1,25 +1,32 @@
 import React, { Component } from 'react';
-import { ScrollView, StyleSheet, View, Text, Button, ImageBackground, TouchableOpacity } from 'react-native';
+import { ScrollView, StyleSheet, View, Text, Image, ImageBackground, TouchableOpacity } from 'react-native';
 import { connect } from 'react-redux';
 
 class DrillPage extends React.Component {
+  _toggleFavorite() {
+    console.log('TOGGLE FAVORITE function ', this.state.drill);
+    const action = { type: 'TOGGLE_FAVORITE', value: this.state.drill };
+    this.props.dispatch(action);
+  }
 
-_toggleFavorite() {
-  const action = { type: "TOGGLE_FAVORITE", value: this.state.dr }
-  this.props.dispatch(action)
-}
+  componentDidUpdate() {
+    console.log('componentDidUpdate : ');
+    console.log(this.props.favoritesDrill);
+  }
 
-componentDidUpdate() {
-  console.log("componentDidUpdate : ")
-  console.log(this.props.favoritesDrill)
-}
+  _displayFavoriteImage() {
+    var sourceImage = require('../Images/ic_favorite_border.png');
+    if (this.props.favoritesDrill.findIndex(item => item.id === this.state.drill.id) !== -1) {
```

Comment:
Same here, maybe use the drill present in the props
` this.props.route.params.drill`